2020/9/encoding/src/main.rs

```diff
@@ -0,0 +1,105 @@
+use std::env;
+use std::fs::File;
+use std::io::{BufRead, BufReader, Error};
+
+fn main() -> Result<(), Error> {
+    let args: Vec<String> = env::args().collect();
+    let reader = BufReader::new(File::open(&args[1])?);
+
+    let input : Vec<_> = reader.lines().map(|line| line.unwrap().parse::<usize>().unwrap()).collect();
```

Comment:
Such functional. Many combinator. So monad.

Reply:
Yeah, really wanted monadic comprehensions over Result and Option so that I could just use ? instead of those two unwraps.  I spent a bit of time playing with the `From` trait, but couldn't get it to work. Looks like there is an unstable feature to help, but oh well.